### PR TITLE
Removing LocalityCheck utility

### LINKED ISSUE #2807: Update Accumulo utilities to extend KeywordExecutable
Must utility applications extend [KeywordExecutable](https://github.com/apache/accumulo/blob/main/start/src/main/java/org/apache/accumulo/start/spi/KeywordExecutable.java). However, there are some that don't and these should be updated to do so. These applications are meant to be used with the `accumulo` command.

I did a search and here is a list of some examples that could be updated as part of this issue. They can be checked off as they get merged to mark as completed.

- [x] [DumpZookeeper](https://github.com/apache/accumulo/pull/2813)
- [ ] [LocalityCheck](https://github.com/apache/accumulo/pull/2808)
- [x] [ChangeSecret](https://github.com/apache/accumulo/pull/2814)
- [x] [DeleteZooInstance](https://github.com/apache/accumulo/pull/2815)
- [x] [RestoreZookeeper](https://github.com/apache/accumulo/pull/2816)
- [x] ~~[TableDiskUsage](https://github.com/apache/accumulo/pull/2809)~~
- [x] [TabletServerLocks](https://github.com/apache/accumulo/pull/2817)
- [x] [VerifyTabletAssignments](https://github.com/apache/accumulo/pull/2818)
- [x] [Cleanup Accumulo Admin utilities](https://github.com/apache/accumulo/pull/2881)

I will create PRs for these and submit them.

### PR DESCRIPTION
**Pull request by cshannon**
Utility is no longer useful so it can be removed.

This closes #2807

### CODE REVIEW

File: server/base/src/main/java/org/apache/accumulo/server/util/LocalityCheck.java [hunk 1]
```diff
@@ -87,7 +102,7 @@ public int run(String[] args) throws Exception {
               blocksForHost);
         }
       }
-      return 0;
+      // return 0;
```
Comment (dlmarion) [inline server/base/src/main/java/org/apache/accumulo/server/util/LocalityCheck.java:105]:
```suggestion
```

**Review by dlmarion — DISMISSED** [review]

**Comment by EdColeman** [comment]
Is this a valid utility to expose to general use?  It seems to address a narrow use-case that might have significant performance issues if arbitrarily run on a large table.  And, other than a developer - does it provide information that someone in general could use?  If someone is worried about file locality, then running a compaction is the only "fix" that I am aware of - and that's just good for a point in time as migrations / splits / ingest occur.  And with external compactions, do they even consider locality?

**Comment by dlmarion** [comment]
I don't think that this PR makes this utility more or less available for general use. It just brings the execution of this utility in line with the others. If you are asking whether or not this utility is useful - maybe it's not.

**Comment by EdColeman** [comment]
Extending KeywordExecutable will display the command as help - when you execute accumulo with no arguments, it shows the available commands - that would raise the visibility of this command - and as mentioned elsewhere, implies a level of correctness and utility that may not be wanted for this command.

**Comment by dlmarion** [comment]
LGTM. There is an open question as to whether or not we should add this utility to the `accumulo` command. Based on the comments above and a discussion with @EdColeman, I am of the opinion that the utility should be removed. As described [here](https://hadoop.apache.org/docs/r3.3.3/hadoop-project-dist/hadoop-hdfs/HdfsDesign.html#Replica_Placement:_The_First_Baby_Steps) the write pipeline will put a replica on the local machine if the writer is on a datanode. Blocks for a file will only be local if a compaction has occurred and *all* of the following are true:

  1. The TabletServer is running on a host with a DataNode
  2. The Tablet has not been re-hosted on a different TabletServer on a different host.
  3. In the case of a External Compaction, the Compactor running the compaction is on the same host as the Tablets' TabletServer.
  4. The Hadoop configuration does not use a  [block placement policy](https://hadoop.apache.org/docs/r3.3.3/hadoop-project-dist/hadoop-hdfs/HdfsBlockPlacementPolicies.html) that does not place a block local to the writer.
  5. Erasure Coding is not being used.
  6. The HDFS balancer has not moved any of the files blocks from the local node.

**Comment by cshannon** [comment]
So should we go ahead close this PR? Reading over everything I agree that it doesn't seem like the utility is useful to expose to everyone and is only helpful for developers based on comments @EdColeman  and @dlmarion .

**Comment by dlmarion** [comment]
I'm of the opinion that we should have a common way for interacting with all utilities. If the utility is not useful, then maybe it should be removed. I don't like the idea of having hidden utilities.

**Comment by cshannon** [comment]
> I'm of the opinion that we should have a common way for interacting with all utilities. If the utility is not useful, then maybe it should be removed. I don't like the idea of having hidden utilities.

Makes sense, if we want to keep the utility it makes sense to me to refactor this and move the utility to the Admin command like the other PRs otherwise just remove it entirely. 

Which direction do we want to go? Do we think the utility has enough use to still keep it around? I don't have enough experience yet to really give an opinion on removal or not.

**Comment by dlmarion** [comment]
@milleruntime @ctubbsii - we have two different opinions on this, we could use some other input here to figure out a way forward.

**Comment by milleruntime** [comment]
> move the utility to the Admin command like the other PRs otherwise just remove it entirely.

I would prefer this be moved under Admin, like we did with the other utilities. This one seems too specific to make it a high level Keyword.

**Comment by dlmarion** [comment]
Seems like a good compromise.

**Comment by cshannon** [comment]
@dlmarion - I went ahead and just started over from main and just removed the file in a new commit and force pushed up as that seemed cleanest since we are just removing the utility now. This is ready for review as it's just a file deletion.

**Review by dlmarion — APPROVED** [review]

**Review by EdColeman — APPROVED** [review]

**Review by milleruntime — APPROVED** [review]

**Pull request merged**